cli/pcluster/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,478 @@
+    Type: CommaDelimitedList
+  # RAID parameters
+  CreateRAIDSubstack:
+    Description: Parameter telling if a RAID Substack has been created
+    Type: String
+  RAIDOptions:
+    Description: Comma Separated List of RAID related options, 8 parameters in total, [shared_dir,raid_type,num_of_raid_volumes,volume_type,volume_size,volume_iops,encrypted,ebs_kms_key_id]
+    Type: CommaDelimitedList
+  RAIDVolumesIDs:
+    Description: Volume IDs of the resulted RAID EBS volumes
+    Type: CommaDelimitedList
+  # EFS parameters
+  CreateEFSSubstack:
+    Description: Parameter telling if an EFS Substack has been created
+    Type: String
+  EFSFileSystemID:
+    Description: ID of the EFS volume used
+    Type: String
+  # FSx parameters
+  CreateFSXSubstack:
+    Description: Parameter telling if a FSx Substack has been created
+    Type: String
+  FSXFileSystemId:
+    Description: ID of the FSx volume used
+    Type: String
+Conditions:
+  # RAID conditions
+  CreateRAIDSubstack: !Equals
+    - !Ref 'CreateRAIDSubstack'
+    - 'true'
+  # Duplicate of the raid-substack code
+  UseRAIDVol1: !Not
+    - !Equals
+      - !Select
+        - '0'
+        - !Ref 'RAIDOptions'
+      - NONE
+  UseRAIDVol2: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '1'
+    - !Condition 'UseRAIDVol1'
+  UseRAIDVol3: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '2'
+    - !Condition 'UseRAIDVol2'
+  UseRAIDVol4: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '3'
+    - !Condition 'UseRAIDVol3'
+  UseRAIDVol5: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '4'
+    - !Condition 'UseRAIDVol4'
+  # EFS conditions
+  CreateEFSSubstack: !Equals
+    - !Ref 'CreateEFSSubstack'
+    - 'true'
+  # FSx conditions
+  CreateFSXSubstack: !Equals
+    - !Ref 'CreateFSXSubstack'
+    - 'true'
+Resources:
+  MasterCloudWatchDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+      DashboardName: !Join
+        - '-'
+        - - !Ref 'PclusterStackName'
+          - 'Master'
+      DashboardBody: !Join
+        - ''
+        - - '{"widgets":['
+          - !Join # TODO: delete this join (since the one before is the same) if we do not change it back
+            - '' # Do not use comma join: Would make a problem when using If statements as Join(',', ['a','','b'])='a,,b'
+            - # Head Node Instance Metrics
+              - '{"type":"text","x":0,"y":2,"width":24,"height":1,"properties":{"markdown":"\n#
+                      Head Node Instance Metrics\n"}}'
+              - !Sub ',{"type":"metric","x":0,"y":3,"width":6,"height":6,"properties":{"view":"timeSeries","stacked":false,"metrics":[["AWS/EC2","CPUUtilization","InstanceId","${MasterInstanceID}"]],"region":"${AWS::Region}"}}'
```

Comment:
it seems to me that the code could be more readable by splitting this sub into two rows: 
one to define the position: `',{"type":"metric","x":0,"y":3,"width":6,"height":6,"properties":{"view":"timeSeries","stacked":false,"metrics":`
and another one to define the metric:
`[["AWS/EC2","CPUUtilization","InstanceId","${MasterInstanceID}"]],"region":"${AWS::Region}"}}'`

Reply:
Right, done also for EFS and FSx